docs/structured-streaming-programming-guide.md

```diff
@@ -1337,7 +1337,7 @@ impressionsWithWatermark.join(
     clickTime >= impressionTime AND
     clickTime <= impressionTime + interval 1 hour
     """),
-  joinType = "leftOuter"      // can be "inner", "leftOuter", "rightOuter"
+  joinType = "leftOuter"      // can be "inner", "leftOuter", "rightOuter", "fullOuter", "leftSemi"
```

Comment:
Not due to this PR, but I saw this section is "Outer Joins with Watermarking", seems to be "inner" and "leftSemi" should not be here.

Reply:
probably the purpose was to enumerate the all possible values for joinType. while I agree it's not matching with the section, I feel it's not a big deal. either is fine, including leaving it as it is (with adding left semi) and another minor PR to fix.

Reply:
I don't feel this is too big deal, leave it as it is for now.